ProjectBank/Infrastructure.Tests/ProjectRepositoryTests.cs

```diff
@@ -21,119 +20,187 @@ public async Task CreateProject_given_CreateProjectDTO_returns_ProjectDTO()
+        await _projectRepository.CreateProjectAsync(new CreateProjectDTO { Title = "Algo", UserId = 1, Description = "Very fun", Tags = new List<string> { "Sorting" } });
+        await _projectRepository.CreateProjectAsync(new CreateProjectDTO { Title = "DMAT", UserId = 2, Description = "Very very fun", Tags = new List<string> { "Math" } });
+        await _projectRepository.EditProjectAsync(new UpdateProjectDTO { Id = 1, Title = "Something else than Algo", Description = "Not sorting", UserId = 1, Tags = new List<string> { "Economy", "Math" } });
+        await _projectRepository.EditProjectAsync(new UpdateProjectDTO { Id = 2, Title = "DMAT", UserId = 2, Description = "Very very fun" });
+        var editedProjectWithTags = await _context.Projects.FindAsync(1);
+        var editedProjectWithoutTags = await _context.Projects.FindAsync(2);
+
+        Assert.Equal(1, editedProjectWithTags.Id);
+        Assert.Equal("Something else than Algo", editedProjectWithTags.Title);
+        Assert.Equal(Status.Active, editedProjectWithTags.Status);
+        Assert.Equal(1, editedProjectWithTags.UserId);
+        Assert.Equal("Not sorting", editedProjectWithTags.Description);
+        Assert.Equal(DateTime.UtcNow, editedProjectWithTags.UpdatedDate, precision: TimeSpan.FromSeconds(5));
+        Assert.Equal(new List<string>() { "Economy", "Math" }, editedProjectWithTags.Tags.Select(tag => new string(tag.Name)).ToList());
+        Assert.Equal(new List<string>(), editedProjectWithoutTags.Tags.Select(tag => new string(tag.Name)).ToList());
     }
 
     [Fact]
-
     public async Task ReadAll_returns_list_of_ProjectDTO()
     {
-        await _projectRepository.CreateProjectAsync(new CreateProjectDTO { Title = "Algo", UserId = 1 });
-        await _projectRepository.CreateProjectAsync(new CreateProjectDTO { Title = "DMAT", UserId = 2 });
-        await _projectRepository.CreateProjectAsync(new CreateProjectDTO { Title = "Disys", UserId = 3 });
+        await _projectRepository.CreateProjectAsync(new CreateProjectDTO { Title = "Algo", UserId = 1, Description = "Very fun", Tags = new List<string> { "Sorting" } });
+        await _projectRepository.CreateProjectAsync(new CreateProjectDTO { Title = "DMAT", UserId = 2, Description = "Very very fun", Tags = new List<string> { "Math" } });
+        await _projectRepository.CreateProjectAsync(new CreateProjectDTO { Title = "Disys", UserId = 3, Description = "Very very very fun", Tags = new List<string> { "Servers" } });
 
         var projects = await _projectRepository.ReadAllAsync();
 
-        Assert.Collection(projects,
-        project => Assert.Equal(new ProjectDTO(1, "Algo", Status.Active, 1), project),
-        project => Assert.Equal(new ProjectDTO(2, "DMAT", Status.Active, 2), project),
-        project => Assert.Equal(new ProjectDTO(3, "Disys", Status.Active, 3), project)
-        );
+        //TODO fix this to assert collection sometime
```

Comment:
Could we maybe have arrays of data and use collections? :/ i haven't tested this, just an idea
```suggestion
        //TODO fix this to assert collection sometime
        
        //PSEUDOCODE
        var names = ["Algo","DMAT","Disys];
        Assert.Collection(projects,
            project => Assert.Equal(new ProjectDTO(1, names[0], [...]),
            project => Assert.Equal(new ProjectDTO(2, names[1], [...]),
            project => Assert.Equal(new ProjectDTO(3, names[2], [...])
        );
```

Reply:
Then we can avoid the project0, project1, [...] 🍝 ❤️